deploy/crds/mattermost_v1alpha1_clusterinstallation_crd.yaml

```diff
@@ -102,6 +102,31 @@ spec:
                     is blue or green.
                   type: string
               type: object
+            canary:
+              properties:
+                canaryDeployment:
+                  description: CanaryDeployment defines the canary deployment.
+                  properties:
+                    image:
+                      description: Image defines the base Docker image that will be
+                        used for the deployment. Required when BlueGreen is enabled.
```

Comment:
We should probably update the comment on this field in the AppDeployment struct so that it doesn't mention BlueGreen now that it's being used for other things as well.

Reply:
@jwilander I was thinking of adding Canary as well. For example, `Required when BlueGreen or Canary is enabled`. This way the user will know that it is a required option when either of the two deployment types is used.